### PR TITLE
Jax support 

### PR DESCRIPTION
**Pull request by MHashemzadeh**
Basic MLP and convolutional neural networks are created.

### CODE REVIEW

File: hive/agents_jax/qnets/base.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    files or command line arguments.
+    """
+
+    @classmethod
+    def type_name(cls):
+        """
+        Returns:
+            "function"
+        """
+        return "function"
```
Comment (dapatil211) [inline hive/agents_jax/qnets/base.py]: Delete

File: hive/agents_jax/qnets/conv.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+import gin
+import jax
+import jax.numpy as jnp
+import numpy as onp
+
+from hive.agents.qnets.mlp import MLPNetwork
+from hive.agents.qnets.utils import calculate_output_dim
+
+
+@gin.configurable
```
Comment (dapatil211) [inline hive/agents_jax/qnets/conv.py]: delete

File: hive/agents_jax/qnets/conv.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
+import torch
+from torch import nn
+
+from typing import Tuple, Union
+
+from flax import linen as nn
+import gin
```
Comment (dapatil211) [inline hive/agents_jax/qnets/conv.py]: delete

File: hive/agents_jax/qnets/conv.py [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+import jax
+import jax.numpy as jnp
+import numpy as onp
+
+from hive.agents.qnets.mlp import MLPNetwork
```
Comment (dapatil211) [inline hive/agents_jax/qnets/conv.py]: Shouldn't be here, it's torch

File: hive/agents_jax/qnets/conv.py [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+                :py:class:`~hive.agents.qnets.noisy_linear.NoisyLinear` layers or
+                :py:class:`torch.nn.Linear` layers.
+            std_init (float): The range for the initialization of the standard
+                deviation of the weights in
+                :py:class:`~hive.agents.qnets.noisy_linear.NoisyLinear`.
+        """
+        super().__init__()
+        self._normalization_factor = normalization_factor
+
+        initializer = nn.initializers.xavier_uniform()
```
Comment (dapatil211) [inline hive/agents_jax/qnets/conv.py]: should be a parameter, either in this file or dqn file

File: hive/agents_jax/qnets/conv.py [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+                    )
+                )
+                conv_seq.append(nn.relu())
+            self.conv = conv_seq
+        else:
+            self.conv = nn.Identity()  ## TODO check identity in flax
+
+        if mlp_layers is not None:
+            # MLP Layers
+            conv_output_size = calculate_output_dim(self.conv, in_dim)
```
Comment (dapatil211) [inline hive/agents_jax/qnets/conv.py:96]: You'll probably need to write a jax version

File: hive/agents_jax/qnets/conv.py (hunk 4 above)
Comment (dapatil211) [reply hive/agents_jax/qnets/conv.py]: Should be jax version

**Review by dapatil211 — CHANGES_REQUESTED** [review]

File: hive/agents_jax/qnets/conv.py (hunk 3 above)
Comment (MHashemzadeh) [reply hive/agents_jax/qnets/conv.py]: It is deleted

**Pull request closed without merging**